src/dotnet/APIView/APIViewWeb/Repositories/ReviewManager.cs

```diff
@@ -181,22 +183,33 @@ internal async Task UpdateReviewAsync(ClaimsPrincipal user, string id)
             await _notificationManager.NotifySubscribersOnNewRevisionAsync(revision, user);
         }
 
-        private async Task<ReviewCodeFileModel> CreateFileAsync(string revisionId, string originalName, Stream fileStream, bool runAnalysis)
+        private async Task<ReviewCodeFileModel> CreateFileAsync(
+            string revisionId, 
+            string originalName,
+            string label,
+            Stream fileStream,
+            bool runAnalysis)
         {
             var originalNameExtension = Path.GetExtension(originalName);
             var languageService = _languageServices.Single(s => s.IsSupportedExtension(originalNameExtension));
 
-            var reviewCodeFileModel = new ReviewCodeFileModel();
-            reviewCodeFileModel.HasOriginal = true;
-            reviewCodeFileModel.Name = originalName;
+            var reviewCodeFileModel = new ReviewCodeFileModel
+            {
+                HasOriginal = true,
+                RevisionNumberSeparateFromName = true
+            };
 
             using (var memoryStream = new MemoryStream())
             {
                 await fileStream.CopyToAsync(memoryStream);
 
                 memoryStream.Position = 0;
 
-                CodeFile codeFile = await languageService.GetCodeFileAsync(originalName, memoryStream, runAnalysis);
+                CodeFile codeFile = await languageService.GetCodeFileAsync(
+                    originalName,
+                    memoryStream,
+                    runAnalysis);
+                codeFile.Name = label != null ? $"{label} - {codeFile.Name}" : codeFile.Name;
```

Comment:
I don't like how we are mixing name and label. Is there a reason you didn't add an additional property?

Reply:
I figured users may want to edit the name/label for revisions. For instance, they may not want every revision to have the same assembly name repeated over and over, even though we default to this.

Reply:
Are we sure we want that?

Reply:
Well, it was shown in the write up for this https://github.com/Azure/azure-sdk-tools/issues/782, though it is not explicitly called out.

Reply:
I would be fine with restricting this to just being able to edit the label, though I think the way it is now gives users more flexibility at little cost.

Reply:
I would always keep the default metadata (name/version) and add labels on top. If we think name is too cumbersome we can remove it all at once later.

Reply:
>I think the way it is now gives users more flexibility at little cost

I'm worried about inconsistent experience if everyone would start setting names to whatever they want more than development cost.